core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -536,6 +538,12 @@ trait EntityComponent {
 
         val sourceShardId = determineShard(clonedWorkspaceId)
         val destShardId = determineShard(newWorkspaceId)
+
+        cloneEntitiesToNewWorkspace(clonedWorkspaceId, newWorkspaceId).map({p =>
+          clonedWorkspaceEntityHistogram += p._1
+          clonedWorkspaceAttributeHistogram += p._2
+        })
```

Comment:
oooh calling `cloneEntitiesToNewWorkspace` a second time here would cause problems. `cloneEntitiesToNewWorkspace` actually performs the database inserts - so adding it here would mean that for every workspace-clone operation the user requests, we copy every single entity and attribute TWICE - obviously we don't want that.

I should have been clearer about my suggestion earlier … I think that instead of updating the histogram values here, you should delete this block of code and move the
```
          clonedWorkspaceEntityHistogram += p._1
          clonedWorkspaceAttributeHistogram += p._2
```
lines inside the `yield` clause of `cloneEntitiesToNewWorkspace` (changing `p._1` to `entitiesCopiedCount` and `p._2` to `attributesCopiedCount`). So, you're updating the histograms right at the code location where the copying happens.